controller/eks-cluster-config-handler.go

```diff
@@ -147,6 +147,11 @@ func (h *Handler) OnEksConfigRemoved(key string, config *eksv1.EKSClusterConfig)
 		logrus.Infof("cluster [%s] is imported, will not delete EKS cluster", config.Name)
 		return config, nil
 	}
+	if config.Status.Phase == eksConfigNotCreatedPhase {
+		// The most likely context here is that the cluster already existed in EKS, so we shouldn't delete it
+		logrus.Infof("cluster [%s] never advanced to creating status, will not delete EKS cluster")
+		return config, nil
+	}
```

Comment:
Assuming this would also work with imported. Wouldn't we also want to show this message to user in the UI ?

Reply:
Once a cluster is deleted, it is removed from the UI pretty quickly. I believe that this message would only flash on the UI for a moment unless something in the UI changed.

Reply:
Makes sense, @rmweir maybe you have another idea. We could run into a scenario where someone deletes the cluster and it sticks around without them knowing, that costs money. If nothing else we should log. Maybe bump to warn ?

Reply:
Just a thought: the message that is displayed in the UI could be changed to: "cannot create cluster [%s] because a cluster in EKS exists with the same name, removing the cluster from Rancher will have no effect on the existing one"

Reply:
@cbron makes a good point though, maybe the cluster has the same name but we still brought up a vpc for it and in that case should we tear it down? would require some more code since it doesn't really differentiate if it exists or it had to create it right now. Easiest way would be to not remove anything but communicate that.

Reply:
I understand what you are saying: we don't want to waste any money. Right now, we check for an existing cluster before anything is created. Is it enough to maintain this going forward?

Reply:
I think that should prevent this.

Reply:
I think that should handle this, we may run into situation in the future where it doesn't get into "creating" for other reasons like an aws limit, so we should keep that in mind at least.

Reply:
nit: maybe we should leave a comment here explaining